### PR TITLE
Makeover flake8

### PR DESCRIPTION
**Pull request by safl**
Adding flake8 to self-test and fixed what it found except for three annotation issues.

### CODE REVIEW

File: .github/workflows/selftest.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install graphviz wheel pylint
+        pip install graphviz wheel flake8
```
Comment (micvbang) [inline .github/workflows/selftest.yml:38]: I think it would make sense to move `flake8` up to the line above

File: testcases/cijoe_flake8.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+paths=""
+for path in {bin,hooks,testcases}/* "${CIJ_PKG_REPOS}/modules/cij/"*.py "${CIJ_PKG_REPOS}/setup.py"; do
+  if [[ -d "$path" ]]; then
+    continue
+  fi
+
+  if grep "^#.*python" "$path" > /dev/null || [[ $path == *.py ]]; then
+    paths="$paths $path"
+  fi
+done
```
Comment (micvbang) [inline testcases/cijoe_flake8.sh:32]: This is bash voodoo magic

File: modules/cij/__init__.py [hunk 3]
```diff
@@ -110,13 +111,15 @@ def expand_path(path):
 
     conf = {v: os.environ.get("_".join([prefix, v])) for v in names}
 
-    for env in (e for e in conf.keys() if e[:len(prefix)] in names and conf[e]):
+    for env in (e for e in conf.keys()
+                if e[:len(prefix)] in names and conf[e]):
         conf[env] = expand_path(conf[env])
         if not os.path.exists(conf[env]):
```
Comment (micvbang) [inline modules/cij/__init__.py:117]: I think the readability of this filter would be improved if it was assigned to a variable and given a name

File: bin/cij_fetch [hunk 4]
```diff
@@ -82,12 +85,14 @@ def meta_reservoir(url):
...
+
 def meta_is_equiv(one, other):
     """Return True two different metas has the same size and md5sum."""
 
     return (one is not None and other is not None) and \
-            (one["length"] == other["length"]) and \
-            (one["md5"] == other["md5"])
+        (one["length"] == other["length"]) and \
+        (one["md5"] == other["md5"])
+
```
Comment (micvbang) [inline bin/cij_fetch:96]: To long expressions like this one, I usually add a parenthesis around the whole thing. It allows one to skip the line continuations, e.g.

```
return (
  (one is not None and other is not None) and
  (one["length"] == other["length"]) and
  (one["md5"] == other["md5"])
)
```

File: modules/cij/runner.py [hunk 5]
```diff
@@ -416,7 +430,8 @@ def trun_to_junitfile(trun, fpath=None):
                     doc_failure = doc.createElement("failure")
                     doc_failure.setAttribute(
                         "message",
-                        "not executed" if tcase.rcode is None else "test failed"
+                        "not executed" if tcase.rcode is None
+                        else "test failed"
```
Comment (micvbang) [inline modules/cij/runner.py:434]: I would personally indent the `else` to align with the `if`

File: modules/cij/runner.py [hunk 6]
```diff
@@ -576,10 +591,10 @@ def tsuite_setup(trun, declr, enum) -> TestSuite:
     tcase_fpaths = []                               # Load testcase fpaths
     if os.path.exists(suite.fpath):              # From suite-file
         suite_lines = (
-            l.strip() for l in open(suite.fpath).read().splitlines()
+            line.strip() for line in open(suite.fpath).read().splitlines()
```
Comment (micvbang) [inline modules/cij/runner.py:594]: The opened file is never closed - consider doing something like

```
suite_content = ""
with open(suite.fpath) as suitef:
  suite_content = suitef.read()
  
suite_lines = ( .... )
  
```

File: bin/cij_tlint [hunk 7]
```diff
@@ -60,8 +60,12 @@ def tsuitefile_testcases_exists(ident, roots, fnames):
 
     for ts_fname in fnames["TSUITES"]:    # Search .suite files
         ts_fpath = os.sep.join([roots["TSUITES"], ts_fname])
-        ts_lines_all = (l.strip() for l in open(ts_fpath).read().splitlines())
-        ts_lines = (l for l in ts_lines_all if len(l) > 1 and l[0] != "#")
+        ts_lines_all = (
+            line.strip() for line in open(ts_fpath).read().splitlines()
```
Comment (micvbang) [inline bin/cij_tlint:64]: Same as above -- this file is never closed

File: bin/cij_tlint [hunk 8]
```diff
@@ -83,8 +87,12 @@ def testcase_unused(ident, roots, fnames):
 
     for ts_fname in fnames["TSUITES"]:  # Testcases from suite-files
         ts_fpath = os.sep.join([roots["TSUITES"], ts_fname])
-        ts_lines_all = (l.strip() for l in open(ts_fpath).read().splitlines())
-        ts_lines = (l for l in ts_lines_all if len(l) > 1 and l[0] != "#")
+        ts_lines_all = (
+            line.strip() for line in open(ts_fpath).read().splitlines()
```
Comment (micvbang) [inline bin/cij_tlint:91]: Same as above -- this file is never closed

**Review by micvbang — COMMENTED** [review]
Looks good - here are some inline comments

File: modules/cij/runner.py (hunk 6 above)
Comment (safl) [reply modules/cij/runner.py]: Good point, thanks

File: modules/cij/runner.py (hunk 5 above)
Comment (safl) [reply modules/cij/runner.py]: I am going with:
```
 "not executed" if tcase.rcode is None else
 "test failed"
```
Linters to not appreciate the long indent to match up to ``if``

File: bin/cij_fetch (hunk 4 above)
Comment (safl) [reply bin/cij_fetch:98]: Much nicer, thanks.

File: modules/cij/__init__.py (hunk 3 above)
Comment (safl) [reply modules/cij/__init__.py]: Agree, I've adjusted accrodingly.

File: .github/workflows/selftest.yml (hunk 1 above)
Comment (safl) [reply .github/workflows/selftest.yml]: Agree, fixed.

File: testcases/cijoe_flake8.sh (hunk 2 above)
Comment (safl) [reply testcases/cijoe_flake8.sh:32]: Yeah, no way around that :)

**Comment by safl** [comment]
Update with changes according to feedback.

File: modules/cij/runner.py [hunk 9]
```diff
@@ -430,8 +430,8 @@ def trun_to_junitfile(trun, fpath=None):
                     doc_failure = doc.createElement("failure")
                     doc_failure.setAttribute(
                         "message",
-                        "not executed" if tcase.rcode is None
-                        else "test failed"
+                        "not executed" if tcase.rcode is None else
+                        "test failed"
```
Comment (micvbang) [inline modules/cij/runner.py:434]: Ah sorry, I wasn't clear enough on this one; I meant

```python
"not executed" if tcase.rcode is None
                        else "test failed"
```
but the version you just posted actually works quite well!

**Comment by micvbang** [comment]
LGTM

**Pull request merged**